### PR TITLE
PLT-14 - User Endpoints - GET Method Added

### PR DESCRIPTION
**Pull request by abdulrkalam**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.service.UserService
+
+@RestController
+@Validated
+@RequestMapping("/person-on-probation-user", produces = [MediaType.APPLICATION_JSON_VALUE, MediaType.IMAGE_JPEG_VALUE])
```
Comment (JamesBoobier) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt:20]: Should these endpoint be protected behind a role?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+class UserResourceController(private val userService: UserService) {
+
+  @GetMapping("/user/{crnId}", produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (JamesBoobier) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt:23]: I think this should just be `crn` not `crnId` to be consistent with other services

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/integration/IntegrationTestBase.kt [hunk 3]
```diff
@@ -16,8 +20,26 @@ abstract class IntegrationTestBase : TestBase() {
...
+    val hmppsAuthMockServer = HmppsAuthMockServer()
+  }
+
   init {
     // Resolves an issue where Wiremock keeps previous sockets open from other tests causing connection resets
     System.setProperty("http.keepAlive", "false")
   }
+
+  protected fun setAuthorisation(
+    user: String = "RESETTLEMENTPASSPORT_ADM",
```
Comment (JamesBoobier) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/integration/IntegrationTestBase.kt:38]: User should PERSONONPROBATION_ADM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt (hunk 2 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt:23]: I will do the change and let  you know when done!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt (hunk 1 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt:20]: Checked with @tomhalson:  yes it should be protected for now we will use this role, later we will have a separate role.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/integration/IntegrationTestBase.kt (hunk 3 above)
Comment (abdulrkalam) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/integration/IntegrationTestBase.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt (hunk 2 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/resource/UserResourceController.kt]: Done

**Pull request merged**